### ISSUE TITLE
Sidebar stuck in overlay mode when page width is increased

### ISSUE THREAD

**Issue by rijenkii** [asyncapi/asyncapi-react#908]
**Description**

When opening the sidebar through the hamburger button on bottom right of the screen, and then increasing the width of the preview, sidebar is still drawn in the overlay mode.

Reproduced on Firefox 121 and Chromium 120.0.6099.216, Fedora 39.

**Steps to reproduce**

Paste the following into the [playground](https://asyncapi.github.io/asyncapi-react/):

<details><summary>Schema</summary>

```json
{
  "asyncapi": "3.0.0",
  "info": {
    "title": "Reproduction",
    "version": "1.0.0"
  },
  "operations": {
    "test": {
      "action": "receive",
      "channel": { "$ref": "#/channels/test" }
    }
  },
  "channels": {
    "test": {
      "address": "/test",
      "messages": {
        "test": { "$ref": "#/components/messages/test" }
      }
    }
  },
  "components": {
    "messages": {
      "test": {
        "payload": {
          "type": "object",
          "properties": {
            "test": {
              "description": "test",
              "type": "string",
              "pattern": "^(?:helloworld)+$"
            }
          },
          "required": ["test"]
        }
      }
    }
  }
}
```

</details>

<details><summary>Configuration</summary>

```json
{
  "show": {
    "sidebar": true,
    "info": true,
    "operations": true,
    "servers": true,
    "messages": true,
    "schemas": true,
    "errors": true
  },
  "expand":{
    "messageExamples": false
  },
  "sidebar": {
    "showServers": "byDefault",
    "showOperations": "byDefault"
  }
}
```

</details>

Click on the hamburger button.
Increase width of the preview side of the screen.

**Expected result**

Overlay sidebar is closed, regular sidebar is drawn instead.

**Actual result**

[Screencast from 2024-02-03 14-53-07.webm](https://github.com/asyncapi/asyncapi-react/assets/5338332/45fca48c-642d-4f9e-8ab5-957787089eb9)

**Troubleshooting**

N/A

**Comment by rijenkii** [comment]
Still an issue.

**Comment by AceTheCreator** [comment]
You mean something like this? @rijenkii ?

https://github.com/user-attachments/assets/eaad5a4e-bc6f-4520-bb0e-53ef0a070050

**Comment by rijenkii** [comment]
Click the hamburger button and expand the right panel all the way out, to the breakpoint that shows the regular sidebar. You haven't expanded it far enough on the video. Reproduction from OP still works, just like the video in the OP.

**Comment by AceTheCreator** [comment]
You mean something like this? @rijenkii 
https://github.com/user-attachments/assets/e20adf82-4105-40f3-bd7d-1086a18abbfb

**Comment by rijenkii** [comment]
Yes, something like this. The bug is still reproducible on https://asyncapi.github.io/asyncapi-react/ though:
<video src="https://github.com/user-attachments/assets/6f095595-91f1-4254-81cb-85f164e771ca"></video>

**Comment by rijenkii** [comment]
Fixed in v2.4.3.
